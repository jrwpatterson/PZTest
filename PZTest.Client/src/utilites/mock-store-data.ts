import { RootReducer } from '../root-reducer'

export const defaultMockStore: RootReducer = {
  basketReducer: {
    basket: {
      id: 'testID',
      lines: [
        {
          grams: 200,
          name: 'test product',
          productID: 'testProdID',
          rowNo: 0,
          price: 5,
        },
        {
          grams: 400,
          name: 'test product 2',
          productID: 'testProdID2',
          rowNo: 1,
          price: 20,
        },
      ],
      total: 25,
    },
    validBasket: true,
  },
  cheeseReducer: {
    cheeses: [
      {
        cheeseColour: 'purple',
        id: 'testid',
        name: 'testCheese',
        pictureUrl: 'testUrl',
        pricePerKG: 10,
      },
      {
        cheeseColour: 'pink',
        id: 'testid2',
        name: 'testPinkCheese',
        pictureUrl: 'testPinkUrl',
        pricePerKG: 15,
      },
    ],
  },
}
