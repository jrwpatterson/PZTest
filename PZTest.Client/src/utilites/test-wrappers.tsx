import React, { ReactType, ReactNode } from 'react'
import { Provider } from 'react-redux'
import configureStore, { MockStoreEnhanced } from 'redux-mock-store'
import { defaultMockStore } from './mock-store-data'
import { render } from '@testing-library/react'
import { renderHook, RenderHookOptions } from '@testing-library/react-hooks'
import { RootReducer } from '../root-reducer'
// import { createMount } from '@material-ui/core/test-utils'
import { Provider as FetchProvider } from 'use-http'
import { MemoryRouter } from 'react-router-dom'

const mockStore = configureStore([])

const createStore = (storeData: RootReducer) => {
  return mockStore(storeData)
}

export const createMockProvider = (
  storeData: RootReducer = defaultMockStore,
) => {
  return ({ children }: { children?: ReactNode }) => (
    <FetchProvider url='http://localhost:5001/'>
      <Provider store={createStore(storeData)}>{children}</Provider>
    </FetchProvider>
  )
}

export const createMockProviderWithStore = (
  fakeStore: MockStoreEnhanced<unknown, {}>,
) => {
  return ({ children }: { children?: ReactNode }) => (
    <FetchProvider url='http://localhost:5001/'>
      <Provider store={fakeStore}>{children}</Provider>
    </FetchProvider>
  )
}

export const renderReduxConnectedComponent = (
  children: ReactType | JSX.Element,
  storeOverride: RootReducer = defaultMockStore,
  initialEntries?: string[],
) => {
  const storeData = storeOverride || defaultMockStore
  // const mount = createMount()
  const store = createStore(storeData)
  const Wrapper = createMockProviderWithStore(store)
  const element = (
    <MemoryRouter initialEntries={initialEntries}>
      <Wrapper>{children}</Wrapper>
    </MemoryRouter>
  )
  const component = render(element)

  return {
    store,
    ...component,
  }
}

export const renderReduxConnectedHook = <T extends {}>(
  callback: (props: unknown) => T,
  storeData: RootReducer = defaultMockStore,
  options: RenderHookOptions<unknown> = {},
) => {
  const store = createStore(storeData)
  const wrapper = createMockProviderWithStore(store)
  const renderedHook = renderHook(callback, { ...options, wrapper })

  return {
    store,
    ...renderedHook,
  }
}
