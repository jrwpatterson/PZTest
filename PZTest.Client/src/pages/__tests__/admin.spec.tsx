import { renderReduxConnectedComponent } from '../../utilites/test-wrappers'
import React from 'react'
import Admin from '../admin'

describe('admin screen', () => {
  it('should match its snapshot', () => {
    const { container } = renderReduxConnectedComponent(<Admin />)
    expect(container).toMatchSnapshot()
  })
})
