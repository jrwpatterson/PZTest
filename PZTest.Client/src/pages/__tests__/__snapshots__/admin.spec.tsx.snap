// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`admin screen should match its snapshot 1`] = `
<div>
  <div
    class="makeStyles-base-1"
  >
    <div>
      Admin Page
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          Cheese Admin Table
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Add New Cheese
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          Name
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          Price/Kg
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          Url
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          Colour
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
        >
          Menu
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <div
          class="MuiPaper-root makeStyles-gridItem-2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          testCheese
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
      >
        <div
          class="MuiPaper-root makeStyles-gridItem-2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          10
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          class="MuiPaper-root makeStyles-gridItem-2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          testUrl
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
      >
        <div
          class="MuiPaper-root makeStyles-gridItem-2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          purple
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <button
          aria-controls="simple-menu"
          aria-haspopup="true"
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-gridItem-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Open Menu
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <div
          class="MuiPaper-root makeStyles-gridItem-2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          testPinkCheese
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
      >
        <div
          class="MuiPaper-root makeStyles-gridItem-2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          15
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div
          class="MuiPaper-root makeStyles-gridItem-2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          testPinkUrl
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-1"
      >
        <div
          class="MuiPaper-root makeStyles-gridItem-2 MuiPaper-elevation1 MuiPaper-rounded"
        >
          pink
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <button
          aria-controls="simple-menu"
          aria-haspopup="true"
          class="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-gridItem-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Open Menu
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
